src/toil/common.py

```diff
@@ -661,10 +650,10 @@ def createBatchSystem(config, jobStore=None, userScript=None):
         else:
             raise RuntimeError('Unrecognised batch system: %s' % config.batchSystem)
 
-        if not (config.disableSharedCache or batchSystemClass.supportsWorkerCleanup()):
-            raise RuntimeError('%s currently does not support shared caching.  Use the '
-                               '--disableSharedCache option when running Toil with this batch '
-                               'system.' % config.batchSystem)
+        if not batchSystemClass.supportsWorkerCleanup():
+            raise RuntimeError('%s currently does not support shared caching.  Use Toil versions'
+                               '< 3.2.0 (Legacy Toil) if you want to use this batch system.' %
```

Comment:
`Use Toil version 3.1.6 along with the --disableSharedCache if you want to …`

The notion of what is "legacy" will change over time.